spec/support/authentication_helpers.rb

```diff
@@ -15,7 +15,14 @@ def sign_in(user, groups: [])
     TestShibbolethHeaders.groups = groups
   end
 
+  def clear_authentication
+    TestShibbolethHeaders.user = nil
+    TestShibbolethHeaders.groups = nil
+  end
+
   RSpec.configure do |config|
+    config.before(:each, type: :request) { clear_authentication }
+    config.before(:each, type: :system) { clear_authentication }
```

Comment:
Can you say why we need to sign out the user before every request and system spec?

Reply:
Added a comment to explain.